rrs/tools/rrs_upstream_history.py

```diff
@@ -0,0 +1,240 @@
+
+    parser.add_option("-d", "--debug",
+            help = "Enable debug output",
+            action="store_const", const=logging.DEBUG, dest="loglevel", default=logging.INFO)
+    
+    parser.add_option("--dry-run",
+            help = "Do not write any data back to the database",
+            action="store_true", dest="dry_run", default=False)
+
+    parser.add_option("--recipe",
+            help = "Recipe IDs to operate on",
+            action="store", dest="recipe", default=None)
+
+    options, args = parser.parse_args(sys.argv)
+    logger.setLevel(options.loglevel)
+
+    if options.plan:
+        maintplans = MaintenancePlan.objects.filter(id=int(options.plan))
+        if not maintplans.exists():
+            logger.error('No maintenance plan with ID %s found' % options.plan)
+            sys.exit(1)
+    else:
+        maintplans = MaintenancePlan.objects.filter(updates_enabled=True)
+        if not maintplans.exists():
+            logger.error('No enabled maintenance plans found')
+            sys.exit(1)
+
+    logger.debug("Starting upstream history...")
+
+    lockfn = os.path.join(fetchdir, "layerindex.lock")
+    lockfile = utils.lock_file(lockfn)
+    if not lockfile:
+        logger.error("Layer index lock timeout expired")
+        sys.exit(1)
+    try:
+        origsyspath = sys.path
+
+        for maintplan in maintplans:
+            for item in maintplan.maintenanceplanlayerbranch_set.all():
+                layerbranch = item.layerbranch
+                try:
+                    with transaction.atomic():
+                        sys.path = origsyspath
+
+                        layer = layerbranch.layer
+                        urldir = layer.get_fetch_dir()
+                        repodir = os.path.join(fetchdir, urldir)
+                        layerdir = os.path.join(repodir, layerbranch.vcs_subdir)
+
+                        recipe_files = []
+                        if options.recipe:
+                            recipe_qry = layerbranch.recipe_set.filter(id__in=options.recipe.split(','))
+                        else:
+                            recipe_qry = layerbranch.recipe_set.all()
+                        for recipe in recipe_qry:
+                            file = str(os.path.join(layerdir, recipe.full_path()))
+                            recipe_files.append(file)
+
+                        (tinfoil, d, recipes, tempdir) = load_recipes(layerbranch, bitbakepath,
+                                fetchdir, settings, logger,  recipe_files=recipe_files)
+                        try:
+
+                            if not recipes:
+                                continue
+
+                            utils.setup_core_layer_sys_path(settings, layerbranch.branch.name)
+
+                            for recipe_data in recipes:
+                                set_regexes(recipe_data)
+
+                            history = RecipeUpstreamHistory(layerbranch=layerbranch, start_date=datetime.now())
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The naive datetime objects are treated by many datetime methods as local times, it is preferred to use aware datetimes to represent times in UTC. The recommended way to create an aware datetime object representing a specific timestamp in UTC is by passing `tzinfo` as an argument to the method.

[Learn more](https://docs.python.org/3/library/datetime.html#aware-and-naive-objects)

Similar issue at line number 222.